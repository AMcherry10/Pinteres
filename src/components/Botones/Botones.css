.buttons{
  margin-left: 70px;
  margin-top: 20px;
}

@media (max-width: 700px) {
 .buttons{
   display: none;
 }
}

.paisajes{
  border-radius: 10px;
  color: white !important;
  height: 40px;
  background-color: rgb(173, 116, 226);
  border: 0ch;
  cursor: pointer;
}

.comida{
  border-radius: 10px;
  color: white;
  height: 40px;
  background-color: #DAE36E;
  border: 0ch;
  cursor: pointer;
  margin-left: 5px;
}

.bodas{
  border-radius: 10px;
  color: white;
  height: 40px;
  background-color: #7A7D82;
  border: 0ch;
  cursor: pointer;
  margin-left: 5px;
}

.idiomas{
  border-radius: 10px;
  color: white;
  height: 40px;
  background-color: #99CAD9;
  border: 0ch;
  cursor: pointer;
  margin-left: 5px;
}

.reposteria{
  border-radius: 10px;
  color: white;
  height: 40px;
  background-color: #E99D8B;
  border: 0ch;
  cursor: pointer;
  margin-left: 5px;
}

.decoracion{
  border-radius: 10px;
  color: white;
  height: 40px;
  background-color: #CC1F25;
  border: 0ch;
  cursor: pointer;
  margin-left: 5px;
}

.tecnologia{
  border-radius: 10px;
  color: white;
  height: 40px;
  background-color: #AB91BE;
  border: 0ch;
  cursor: pointer;
  margin-left: 5px;
}

.moda{
  border-radius: 10px;
  color: white;
  height: 40px;
  background-color: #564511c0;
  border: 0ch;
  cursor: pointer;
  margin-left: 5px;
}

.populares{
  border-radius: 10px;
  color: white;
  height: 40px;
  background-color: #8D625B;
  border: 0ch;
  cursor: pointer;
  margin-left: 5px;
}

.regalos{
  border-radius: 10px;
  color: white;
  height: 40px;
  background-color: #E77F64;
  border: 0ch;
  cursor: pointer;
  margin-left: 5px;
}

.arquitectura{
  border-radius: 10px;
  color: white;
  height: 40px;
  background-color: #3B9D90;
  border: 0ch;
  cursor: pointer;
  margin-left: 5px;
}

.logos{
  border-radius: 10px;
  color: white;
  height: 40px;
  background-color: #C18E8A;
  border: 0ch;
  cursor: pointer;
  margin-left: 5px;
}

.btn-primary:hover {
  color: #fff;
  background-color: #e0c3fc !important;
  border-color: transparent !important;
}

.btn-primary:focus{
  outline: 0;
    box-shadow: 0 0 0 0.2rem transparent !important;
}