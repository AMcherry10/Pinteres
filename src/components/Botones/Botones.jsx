import React, { Component } from "react";
import "./Botones.css";

class Botones extends Component {
  actualizarBusqueda = (event)=>{
    this.props.actualizarBusqueda(event.target.name)
  }

  render() {
    return (
      <div>
        <div className="col-12 col-xs-12 col-md-12 col-lg-12 col-xl-12">
          <div className="buttons">
            <a className="btn btn-primary paisajes" name="Paisaje" href="#"  role="button" onClick={this.actualizarBusqueda} >Paisajes</a>
            <a className="btn btn-primary comida" name="Comida" href="#" role="button" onClick={this.actualizarBusqueda}>Comida</a>
            <a className="btn btn-primary bodas" href="#" role="button">Bodas</a>
            <a className="btn btn-primary idiomas" href="#" role="button">Idiomas</a>
            <a className="btn btn-primary reposteria" href="#" role="button">Repostería</a>
            <a className="btn btn-primary decoracion" href="#" role="button">Decoración</a>
            <a className="btn btn-primary tecnologia" href="#" role="button">Tecnología</a>
            <a className="btn btn-primary moda" href="#" role="button">Moda</a>
            <a className="btn btn-primary populares" href="#" role="button">Populares</a>
            <a className="btn btn-primary regalos" href="#" role="button">Regalos</a>
            <a className="btn btn-primary arquitectura" href="#" role="button">Arquitectura</a>
            <a className="btn btn-primary logos" href="#" role="button">Logos</a>
          </div>
        </div>
      </div>
    )
  }
}

export default Botones;