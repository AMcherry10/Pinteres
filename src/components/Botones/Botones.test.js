import React from "react";
import Enzyme, { shallow, mount, render } from "enzyme";
import renderer from "react-test-renderer";
import Botones from "../Botones/Botones";
import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({ adapter: new Adapter() });

describe("Botones Component", () => {
 
  // make our assertion and what we expect to happen 
  it("hace render de componente", () => {
    expect(shallow(<Botones />).find("div.buttons").exists()).toBe(true)
  })

  it("render correcto", () => {
    const component = renderer.create(
      <Botones/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
 })