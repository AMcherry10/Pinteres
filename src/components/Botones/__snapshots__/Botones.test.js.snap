// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Botones Component render correcto 1`] = `
<div>
  <div
    className="col-12 col-xs-12 col-md-12 col-lg-12 col-xl-12"
  >
    <div
      className="buttons"
    >
      <a
        className="btn btn-primary paisajes"
        href="#"
        name="Paisaje"
        onClick={[Function]}
        role="button"
      >
        Paisajes
      </a>
      <a
        className="btn btn-primary comida"
        href="#"
        name="Comida"
        onClick={[Function]}
        role="button"
      >
        Comida
      </a>
      <a
        className="btn btn-primary bodas"
        href="#"
        name="Bodas"
        onClick={[Function]}
        role="button"
      >
        Bodas
      </a>
      <a
        className="btn btn-primary idiomas"
        href="#"
        name="Idiomas"
        onClick={[Function]}
        role="button"
      >
        Idiomas
      </a>
      <a
        className="btn btn-primary reposteria"
        href="#"
        name="Reposteria"
        onClick={[Function]}
        role="button"
      >
        Repostería
      </a>
      <a
        className="btn btn-primary decoracion"
        href="#"
        name="Decoracion"
        onClick={[Function]}
        role="button"
      >
        Decoración
      </a>
      <a
        className="btn btn-primary tecnologia"
        href="#"
        name="Tecnologia"
        onClick={[Function]}
        role="button"
      >
        Tecnología
      </a>
      <a
        className="btn btn-primary moda"
        href="#"
        name="Moda"
        onClick={[Function]}
        role="button"
      >
        Moda
      </a>
      <a
        className="btn btn-primary populares"
        href="#"
        name="Populares"
        onClick={[Function]}
        role="button"
      >
        Populares
      </a>
      <a
        className="btn btn-primary regalos"
        href="#"
        name="Regalos"
        onClick={[Function]}
        role="button"
      >
        Regalos
      </a>
      <a
        className="btn btn-primary arquitectura"
        href="#"
        name="Arquitectura"
        onClick={[Function]}
        role="button"
      >
        Arquitectura
      </a>
      <a
        className="btn btn-primary logos"
        href="#"
        name="Logos"
        onClick={[Function]}
        role="button"
      >
        Logos
      </a>
    </div>
  </div>
</div>
`;
